rust/lance-index/src/vector/flat/storage.rs

```diff
@@ -179,29 +176,43 @@ impl VectorStore for FlatStorage {
     }
 }
 
-pub struct FlatDistanceCal {
-    vectors: Arc<FixedSizeListArray>,
-    query: ArrayRef,
-    distance_type: DistanceType,
+pub struct FlatDistanceCal<'a> {
+    vectors: &'a [f32],
```

Comment:
Let's keep these types be still arrow array cause we need to support binary vectors

Reply:
Later we can optimize this with generic